apm/monitor.go

```diff
@@ -0,0 +1,143 @@
+package apm
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/evergreen-ci/birch"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/event"
+)
+
+type basicMonitor struct {
+	config *MonitorConfig
+
+	inProg     map[int64]eventKey
+	inProgLock sync.Mutex
+
+	current        map[eventKey]*eventRecord
+	currentStartAt time.Time
+	currentLock    sync.Mutex
+}
+
+// NewwBasicMonitor returns a simple monitor implementation that does
+// not automatically rotate data. The MonitorConfig makes it possible to
+// filter events. If this value is nil, no events will be filtered.
+func NewBasicMonitor(config *MonitorConfig) Monitor {
+	return &basicMonitor{
+		config:         config,
+		inProg:         make(map[int64]eventKey),
+		current:        make(map[eventKey]*eventRecord),
+		currentStartAt: time.Now(),
+	}
+}
+
+func (m *basicMonitor) popRequest(id int64) eventKey {
+	m.inProgLock.Lock()
+	defer m.inProgLock.Unlock()
+
+	out, ok := m.inProg[id]
+	if ok {
+		delete(m.inProg, id)
+	}
+
+	return out
+}
+
+func (m *basicMonitor) setRequest(id int64, key eventKey) {
+	if !m.config.shouldTrack(key) {
+		return
+	}
+
+	m.inProgLock.Lock()
+	defer m.inProgLock.Unlock()
+
+	m.inProg[id] = key
+}
+
+func (m *basicMonitor) getRecord(id int64) *eventRecord {
+	key := m.popRequest(id)
+	if key.isNil() {
+		return nil
+	}
+
+	m.currentLock.Lock()
+	defer m.currentLock.Unlock()
+
+	event := m.current[key]
+	if event == nil {
```

Comment:
shouldn't we avoid using the var name `event` since it is also the name of an imported package?